server.cpp

```diff
@@ -303,11 +309,53 @@ void proxy_remote_file(struct server_app *app, int client_socket, const char *re
     // Bonus:
     // * When connection to the remote server fail, properly generate
     // HTTP 502 "Bad Gateway" response
+    //std::cout<<"WBHFDOQHGFD)QO)\n";
+    
+    const int PORT= app->remote_port;
+    int client_fd, new_socket;
+    const char *ipAddressStr = app->remote_host;
+    struct in_addr their_in_addr;
+    struct sockaddr_in my_addr; /* my address */
+    struct sockaddr_in their_addr; /* connector addr */ 
+    int sin_size;
+    char buffer[BUFFER_SIZE];
+    char buffer2[BUFFER_SIZE];
+    strcpy(buffer, request.c_str());
 
-    char response[] = "HTTP/1.0 501 Not Implemented\r\n\r\n";
-    send(client_socket, response, strlen(response), 0);
-}
+    if ((client_fd = socket(PF_INET, SOCK_STREAM, 0)) == 0) {
+        perror("socket failed");
+        exit(EXIT_FAILURE);
+    }
 
+    their_addr.sin_family = AF_INET; 
+    their_addr.sin_port = htons(PORT);
```

Comment:
isn't htons supposed to take a hostname as an argument? Maybe this is what you're doing, I'm not entirely sure

Reply:
htons just changes endianness for the port to make it server ready
[Uploading CamScanner 10-20-2023 23.59.pdf…]()

Reply:
I see